### DISCUSSION TITLE
Incremental Static Regeneration not working on my website.

### DISCUSSION THREAD

**Discussion by richardingwe** [vercel/next.js#29195] [Help]
### What version of Next.js are you using?

10.2.3

### What version of Node.js are you using?

v15.12.0

### What browser are you using?

chrome

### What operating system are you using?

Windows

### How are you deploying your application?

Vercel

### Describe the Bug

I built a portfolio website using next.js and sanity headless cms, so on some pages, I used the getStaticProps function and added revalidate to it.

here is is a sample code.

```
export async function getStaticProps() {
    try {
        const data = await sanityClient.fetch(
            `*[_type == "CodeProject"]{
                title,
                slug,
                date,
                place,
                description,
                projectType,
                link,
                githubLink,
                technologiesUsed,
                projectImage{
                asset->{
                    _id,
                    url
                },
                alt
                }
            }`
        );
        return {
            props: {
                codeData: data,
                revalidate: 60,
            },
        };
    } catch (error) {
        console.log(error);
    }

}
```

but Incremental static regeneration is not working. whenever I make changes from sanity cms, I always have to rebuild the site. because no matter how I reload the website does not change.

what exactly am I doing wrong please?
a link to the website [https://www.ruingwe.com](https://www.ruingwe.com)

### Expected Behavior

I expect ISR to make changes on the website without me having to manually redeploy the site

### To Reproduce

```

export async function getStaticProps() {
    try {
        const data = await sanityClient.fetch(
            `*[_type == "CodeProject"]{
                title,
                slug,
                date,
                place,
                description,
                projectType,
                link,
                githubLink,
                technologiesUsed,
                projectImage{
                asset->{
                    _id,
                    url
                },
                alt
                }
            }`
        );
        return {
            props: {
                codeData: data,
                revalidate: 60,
            },
        };
    } catch (error) {
        console.log(error);
    }

}```

**Comment by gu-stav** [comment]
Irecommend updating to https://github.com/vercel/next.js/releases/tag/v11.1.0 which includes many fixes for ISR. Before I ran into the same problems as you.

**Comment by richardingwe** [comment]
> Irecommend updating to https://github.com/vercel/next.js/releases/tag/v11.1.0 which includes many fixes for ISR. Before I ran into the same problems as you.

okay, I just upgraded and it is not still working.

**Comment by gu-stav** [comment]
The second thing which might happen to you and which isn't related to next.js: the sanity client might cache responses ...

**Comment by richardingwe** [comment]
> The second thing which might happen to you and which isn't related to next.js: the sanity client might cache responses ...

can you please elaborate?

by the way, do you think it being a pwa is a problem?

**Comment by balazsorban44** [comment]
I think being a PWA might be the issue, yes. as mentioned above, your client will cache responses and only a redeploy/reinstall(?) of the PWA will purge that cache correctly.


I'm guessing the site in question is https://github.com/richardingwe/rui-portfolio


Looks like you have to configure the runtime cache behavior. 

https://github.com/shadowwalker/next-pwa#runtime-caching

The default will cache most things for 24 hours, even if you set 60 seconds in getStaticProps:
https://github.com/shadowwalker/next-pwa/blob/master/cache.js
